src/payment/strategies/paypal-commerce/paypal-commerce-credit-card-payment-strategy.ts

```diff
@@ -0,0 +1,91 @@
+import PaymentActionCreator from '../../payment-action-creator';
+import PaymentMethodActionCreator from '../../payment-method-action-creator';
+import { PaymentInitializeOptions, PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+import { DisableFundingType, PaypalCommerceHostedForm, PaypalCommerceInitializationData, PaypalCommerceScriptLoader, PaypalCommerceScriptOptions } from './index';
+
+export default class PaypalCommerceCreditCardPaymentStrategy implements PaymentStrategy {
+    constructor(
+        private _store: CheckoutStore,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _paypalScriptLoader: PaypalCommerceScriptLoader,
+        private _paypalCommerceHostedForm: PaypalCommerceHostedForm,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentActionCreator: PaymentActionCreator
+    ) {}
+
+    async initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const state = await this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(options.methodId));
+        const { initializationData } = state.paymentMethods.getPaymentMethodOrThrow(options.methodId);
+
+        const cart = state.cart.getCartOrThrow();
+        const paramsScript = {
+            options: this._getOptionsScript(initializationData, cart),
+            attr: { clientToken: initializationData.clientToken },
+        };
+
+        const paypal = await this._paypalScriptLoader.loadPaypalCommerce(paramsScript, true);
+
+        if (options.paypalcommerce?.form) {
+            await this._paypalCommerceHostedForm.initialize(options.paypalcommerce.form, cart.id, paypal);
+        }
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    async execute(payload: OrderRequestBody, options: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = payload;
+
+        if (!payment) {
+            throw new PaymentArgumentInvalidError(['payment']);
+        }
+
+        await this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
+
+        const { orderId } = await this._paypalCommerceHostedForm.submit();
+
+        const paymentData =  {
+            formattedPayload: {
+                vault_payment_instrument: null,
+                set_as_default_stored_instrument: null,
+                device_info: null,
+                paypal_account: {
+                    order_id: orderId,
+                },
+            },
+        };
+
+        return this._store.dispatch(this._paymentActionCreator.submitPayment({ ...payment, paymentData }));
+    }
+
+    finalize(): Promise<InternalCheckoutSelectors> {
+        return Promise.reject(new OrderFinalizationNotRequiredError());
+    }
+
+    deinitialize(): Promise<InternalCheckoutSelectors> {
+        return Promise.resolve(this._store.getState());
```

Comment:
Does the provider SDK provides you with a method to clean up? i.e.: Braintree provides us with a `teardown` method.

Reply:
No, [there are](https://developer.paypal.com/docs/platforms/checkout/reference/style-guide/#methods) all methods